UnityEngine.AssetBundles/Editor/AssetBundleBrowser/InspectTab/AssetBundleInspectTab.cs

```diff
@@ -223,15 +270,84 @@ public List<string> BundleList
         public void SetBundleItem(InspectTreeItem selected)
         {
             if (selected == null)
+            {
                 m_SingleInspector.SetBundle(null);
+            }
             else
-                m_SingleInspector.SetBundle(selected.bundle, selected.bundlePath);
+            {
+                AssetBundle bundle = this.LoadBundle(selected.bundlePath);
+                m_SingleInspector.SetBundle(bundle, selected.bundlePath);
+            }
         }
 
         [System.Serializable]
         public class InspectTabData
         {
             public string m_BundlePath = string.Empty;
         }
+
+        /// <summary>
+        /// Loads the bundle at the specified path.
+        /// Unloads previously loaded bundles if neccessary.
+        /// </summary>
+        /// <returns>Returns loaded bundle, null if it could not be loaded.</returns>
+        /// <param name="path">Path of bundle to load</param>
+        private AssetBundle LoadBundle(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return null;
+            }
+
+            string bundleName = Path.GetFileNameWithoutExtension(path);
+            AssetBundle bundle = this.GetLoadedBundleByName(bundleName);
+            if (null != bundle)
```

Comment:
This concerns me, but I'm open to the discussion that it may be the preferred behavior. In my old code, the bundle was loaded once you first clicked on the name of the bundle in the tree. It would only be reloaded if you somehow triggered an unload (refresh, change path, etc.). In this code, every time you select an entry in the tree, it reloads that bundle. So any time you clicked back onto a previously selected entry, it would reload the bundle. That sounds slow and non-ideal to me. Was that an intentional change or an accidental side-effect?  This load-once goal was part of the reasoning behind having the InspectTreeItem handle bundle loading. A tree reload would give me freshly loaded bundles (such as after domain reload).